### PR TITLE
Hw 20 connect my sql

### PR DESCRIPTION
**Pull request by tbondare**

### CODE REVIEW

File: src/main/java/taxiapp/jdbc/ManufacturerDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.setLong(3, manufacturer.getId());
+            if (statement.executeUpdate() != 1) {
+                throw new RuntimeException("update != 1");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturers", e);
+        }
+        return oldValue;
```
Comment (YuraBondarenko) [inline src/main/java/taxiapp/jdbc/ManufacturerDaoJdbcImpl.java:87]:
```suggestion
        String query = "UPDATE manufacturers SET name = ?,"
                + "country = ? WHERE id = ? AND deleted = false;";
        try (Connection connection = ConnectionUtil.getConnection()) {
            PreparedStatement statement = connection.prepareStatement(query);
            statement.setString(1, manufacturer.getName());
            statement.setString(2, manufacturer.getCountry());
            statement.setLong(3, manufacturer.getId());
            statement.executeUpdate();           
        } catch (SQLException e) {
            throw new DataProcessingException("Can't update manufacturers", e);
        }
        return manufacture;
```

File: src/main/java/taxiapp/jdbc/ManufacturerDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    @Override
+    public boolean delete(Long manufacturerId) {
+        String query = "UPDATE manufacturers SET deleted = true WHERE id = ? AND deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, manufacturerId);
+            if (statement.executeUpdate() != 1) {
+                return false;
+            }
```
Comment (YuraBondarenko) [inline src/main/java/taxiapp/jdbc/ManufacturerDaoJdbcImpl.java:98]: exucuteUpdate возвращает количество измененных строк
```suggestion
            return statement.executeUpdate() > 0;
```

File: src/main/java/taxiapp/jdbc/ManufacturerDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        try {
+            Long id = resultSet.getObject("id", Long.class);
+            String name = resultSet.getString("name");
+            String country = resultSet.getString("country");
+            return new Manufacturer(id, name, country);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't parse result set", e);
+        }
+    }
+}
```
Comment (YuraBondarenko) [inline src/main/java/taxiapp/jdbc/ManufacturerDaoJdbcImpl.java:115]: здесь можно без блока try

**Review by YuraBondarenko — CHANGES_REQUESTED** [review]

**Review by YuraBondarenko — APPROVED** [review]

File: src/main/java/taxiapp/dao/CarDao.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package taxiapp.dao;
+
+import java.util.List;
+import java.util.Optional;
+import taxiapp.model.Car;
+
+public interface CarDao {
+
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/dao/CarDao.java:8]: redundant empty line

File: src/main/java/taxiapp/dao/CarDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.List;
+import java.util.Optional;
+import taxiapp.db.Storage;
+import taxiapp.lib.Dao;
+import taxiapp.model.Car;
+import taxiapp.model.Driver;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/dao/CarDaoImpl.java:13]: same

File: src/main/java/taxiapp/dao/CarDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    @Override
+    public Car create(Car car) {
+        Storage.addCar(car);
+        return Storage.cars.get(Storage.cars.size() - 1);
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/dao/CarDaoImpl.java:17]: see previous review on the methods of this class

File: src/main/java/taxiapp/dao/DriverDao.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package taxiapp.dao;
+
+import java.util.List;
+import java.util.Optional;
+import taxiapp.model.Driver;
+
+public interface DriverDao {
+
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/dao/DriverDao.java:8]: same

File: src/main/java/taxiapp/dao/DriverDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+import taxiapp.db.Storage;
+import taxiapp.lib.Dao;
+import taxiapp.model.Driver;
+
+@Dao
+public class DriverDaoImpl implements DriverDao {
+    @Override
+    public Driver create(Driver driver) {
+        Storage.addDriver(driver);
+        return Storage.drivers.get(Storage.drivers.size() - 1);
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/dao/DriverDaoImpl.java:14]: same comments as for CarDaoImpl

File: src/main/java/taxiapp/jdbc/ManufacturerDaoJdbc.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
+package taxiapp.jdbc;
+
+import java.util.List;
+import java.util.Optional;
+import taxiapp.model.Manufacturer;
+
+public interface ManufacturerDaoJdbc {
+
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/jdbc/ManufacturerDaoJdbc.java:8]: it is redundant to create the separate DaoJdbc interface
you should already have ManufacturerDao interface

File: src/main/java/taxiapp/jdbc/ManufacturerDaoJdbcImpl.java [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+import java.util.List;
+import java.util.Optional;
+import taxiapp.exception.DataProcessingException;
+import taxiapp.lib.Dao;
+import taxiapp.model.Manufacturer;
+import taxiapp.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoJdbcImpl implements ManufacturerDaoJdbc {
+
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/jdbc/ManufacturerDaoJdbcImpl.java:18]: redundant empty line

File: src/main/java/taxiapp/jdbc/ManufacturerDaoJdbcImpl.java [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers (name, country)"
+                + "VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/jdbc/ManufacturerDaoJdbcImpl.java:25]: in all methods - open statements in try with resources - see checklist

File: src/main/java/taxiapp/model/Car.java [hunk 12]
```diff
@@ -33,4 +42,18 @@ public void setModel(String model) {
     public void setManufacturer(Manufacturer manufacturer) {
         this.manufacturer = manufacturer;
     }
+
+    public void addToDriveListById(Long driverId) {
+        driverList = Storage.drivers.stream()
+                        .filter(d -> d.getId().equals(driverId))
+                        .collect(Collectors.toList());
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/model/Car.java:50]: this should be removed

File: src/main/java/taxiapp/service/CarService.java [hunk 13]
```diff
@@ -0,0 +1,25 @@
+package taxiapp.service;
+
+import java.util.List;
+import java.util.Optional;
+import taxiapp.model.Car;
+import taxiapp.model.Driver;
+
+public interface CarService {
+
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/service/CarService.java:9]: redundant empty line

File: src/main/java/taxiapp/service/CarService.java [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+    Car create(Car car);
+
+    Optional<Car> get(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/service/CarService.java:12]: `Car`, not `Optional<Car>` - in all services

File: src/main/java/taxiapp/service/CarServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+import taxiapp.model.Driver;
+
+@Service
+public class CarServiceImpl implements CarService {
+
+    @Inject
+    DriverDao driverDao;
+
+    @Inject
+    CarDao carDao;
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/service/CarServiceImpl.java:19]: comments on this were added in the previous review

File: src/main/java/taxiapp/service/ManufacturerServiceJdbcImpl.java [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+
+import java.util.List;
+import java.util.Optional;
+import taxiapp.jdbc.ManufacturerDaoJdbc;
+import taxiapp.lib.Inject;
+import taxiapp.lib.Service;
+import taxiapp.model.Manufacturer;
+
+@Service
+public class ManufacturerServiceJdbcImpl implements ManufacturerService {
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/service/ManufacturerServiceJdbcImpl.java:11]: we should not create a separate service for working with jdbc

File: src/main/java/taxiapp/util/ConnectionUtil.java [hunk 17]
```diff
@@ -0,0 +1,23 @@
+package taxiapp.util;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+
+public class ConnectionUtil {
+
```
Comment (KseniiaMakarova) [inline src/main/java/taxiapp/util/ConnectionUtil.java:9]: redundant empty line

File: src/main/resources/init_db.sql [hunk 18]
```diff
@@ -0,0 +1,8 @@
+CREATE SCHEMA `manufacturers_schema` DEFAULT CHARACTER SET utf8 ;
+CREATE TABLE `manufacturers_schema`.`manufacturers` (
+                                                        `id` BIGINT(11) NOT NULL AUTO_INCREMENT,
+                                                        `name` VARCHAR(225) NOT NULL,
+                                                        `country` VARCHAR(225) NOT NULL,
+                                                        `deleted` TINYINT NOT NULL DEFAULT 0,
+                                                        PRIMARY KEY (`id`),
+                                                        UNIQUE INDEX `id_UNIQUE` (`id` ASC) VISIBLE);
```
Comment (KseniiaMakarova) [inline src/main/resources/init_db.sql:8]: add empty line

**Pull request merged**